### PR TITLE
MGMT-9989 - Patching the Node Root CA by hand instead of using MCO

### PR DESCRIPTION
**Pull request by jparrill**
# Refactoring Wait for resources and enhancing the MCO patch over the spoke

This PR involves 2 Changes

## Wait for deployments refactor 
I've deleted the wait_for_deployment.sh in favor of check_resources function which allow us to check different objects apart from deployments and using w/e Kubeconfig you need.

## Node Root CA chain Patched manually on Spoke Clusters
This only applies to Spoke/Edge clusters

This is a big change, we are using this alternative method in order to avoid reboot the Spoke Cluster nodes (which takes 45 Minutes). 

This method should not cause any collateral issue because all the customer workloads will be added after the spoke relocation, so the node will be rebooted in the end.

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

Fixes:
- [MGMT-9989](https://issues.redhat.com/browse/MGMT-9989)
- [MGMT-10249](https://issues.redhat.com/browse/MGMT-10249)

## Type of change

Please select the appropriate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

- Manually tested on Flavio3
- This should pass the CI, don't set label `skip-ci` plz

### CODE REVIEW

File: shared-utils/common.sh [hunk 1]
```diff
@@ -4,6 +4,36 @@
...
+    # This function will copy manually the registry certificates into the OCP nodes and
+    # it will replace the MCO because it takes 45 Minutes in reboot.
+    # Initially no collateral effects on MCO.
+    # This is only affecting the spokes, because we cannot ensure the Hub's RSA Keyfile location
+
+    cluster=${2}
+    i=${3}
+    cp -f ${PATH_CA_CERT} ${SPOKE_SAFE_FOLDER}
+
+    for master in $(echo $(seq 0 $(($(yq eval ".spokes[${i}].[]|keys" ${SPOKES_FILE} | grep master | wc -l) - 1)))); do
```
[the remark this answers is not in the document]
Comment (jparrill) [reply shared-utils/common.sh:17]: I like it.
Implemented

**Review by flaper87 — APPROVED** [review]
/lgtm

**Review by flaper87 — APPROVED** [review]

**Comment by jparrill** [comment]
Tests passed in Flavio3 ( failed on worker)

**Comment by alknopfler** [comment]
Lgtm

**Review by flaper87 — APPROVED** [review]
We should really move away from using ssh :/

**Pull request merged**